File: frogoui/src/main/java/com/frogobox/ui/fireworks/initializers/RotationInitializer.java
package com.frogobox.ui.fireworks.initializers;

import com.frogobox.ui.fireworks.Particle;

import java.util.Random;

public class RotationInitializer implements ParticleInitializer {

    private final int mMinAngle;
    private final int mMaxAngle;

    public RotationInitializer(int minAngle, int maxAngle) {
        mMinAngle = minAngle;
        mMaxAngle = maxAngle;
    }

    @Override
    public void initParticle(Particle p, Random r) {
        p.mInitialRotation = (mMinAngle == mMaxAngle) ? mMinAngle : r.nextInt(mMaxAngle - mMinAngle) + mMinAngle;
    }

}

File: core-ui-android/src/main/java/com/frogobox/ui/fireworks/initializers/RotationInitializer.kt
package com.frogobox.ui.fireworks.initializers

import com.frogobox.ui.fireworks.Particle
import java.util.*

class RotationInitializer(
    private val mMinAngle: Int,
    private val mMaxAngle: Int,
) : ParticleInitializer {

    override fun initParticle(p: Particle?, r: Random?) {
        p!!.mInitialRotation =
            if (mMinAngle == mMaxAngle) mMinAngle.toFloat() else r!!.nextInt(mMaxAngle - mMinAngle) + mMinAngle.toFloat()
    }

}